interview-questions.md

```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a hash in Ruby?
 
-  Your answer:
+  Your answer: A hash in Ruby is a container of data set in a key:value pair. 
 
-  Researched answer:
+  Researched answer: In Ruby a hash is a container/collection of key: value pairs, with the value able to be referenced by its unique key. 
 
 
 
 2. What is a gem?
 
-  Your answer:
+  Your answer: A gem is a piece of code in Ruby. 
 
-  Researched answer:
+  Researched answer: In Ruby a gem is sharable code or library of code functionality. A gem is open source library of code in the Ruby language. 
 
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer: Ruby on Rails is a framework that uses Ruby language. The framework follow MVC pattern to develop a fullstack site. 
 
-  Researched answer:
+  Researched answer: Ruby on Rails is a server-side web app framework written in the Ruby language. Rails is a fullstack framework using the MVC model. The framework provides a default style database, web application, and web service. 
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: A relational database is text-written data stored in a database, able to retrieve. 
 
-  Researched answer:
+  Researched answer: A relational database is a dtatabase with information set in table with rows and columns. There are other types of database structure including: object oriented database, cloud database, distribution database, and NoSQL database. 
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys are unique indentifiers, usually used in databases to track each individual piece of data to be able to refer back. Comes in handy when pieces of data are similar, the primary keys will be different. 
 
-  Researched answer:
+  Researched answer: Primary keys are unique id's that are given to every row of a table in Rails. If there is duplicate information in a table, the primary key or unique id tells the row apart by their id. 
```

Comment:
While your answers are well worded in this section, I would love to see some more on how you have interacted with these concepts as well.